### PR TITLE
[Databricks] Support VNet peering operations for workspaces

### PR DESCRIPTION
**Pull request by bim-msft**
`az databricks workspace vnet-peering`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Databricks

File: src/databricks/azext_databricks/_params.py [hunk 1]
```diff
@@ -49,3 +50,18 @@ def load_arguments(self, _):
...
     with self.argument_context('databricks workspace wait') as c:
         c.argument('workspace_name', options_list=['--name', '-n'], id_part='name', help='The name of the workspace.')
+
+    for scope in ['create', 'show', 'delete', 'update', 'wait']:
+        with self.argument_context('databricks workspace vnet-peering {}'.format(scope)) as c:
+            c.argument('workspace_name', help='The name of the workspace.')
+            c.argument('peering_name', options_list=['--name', '-n'], help='The name of the vnet peering.')
+
+    with self.argument_context('databricks workspace vnet-peering create') as c:
+        c.argument('remote_virtual_network', options_list=['--remote-vnet'], help='The Id of the remote virtual network.')
```
Comment (fengzhou-msft) [inline src/databricks/azext_databricks/_params.py:60]: Can id or name be supported for this parameter?

File: src/databricks/azext_databricks/commands.py [hunk 2]
```diff
@@ -13,18 +13,27 @@
 
 def load_command_table(self, _):
 
-    from ._client_factory import cf_workspaces
-
-    with self.command_group('databricks', is_preview=True):
-        pass
```
Comment (fengzhou-msft) [inline src/databricks/azext_databricks/commands.py:19]: Just to confirm, preview tag removed to target GA?

File: src/databricks/azext_databricks/custom.py [hunk 3]
```diff
@@ -99,3 +99,70 @@ def list_databricks_workspace(cmd, client, resource_group_name=None):
...
+                                   no_wait=False):
+    from .vendored_sdks.databricks.models._models_py3 import \
+        VirtualNetworkPeering, VirtualNetworkPeeringPropertiesFormatRemoteVirtualNetwork
+
+    peering = VirtualNetworkPeering(
+        remote_virtual_network=VirtualNetworkPeeringPropertiesFormatRemoteVirtualNetwork(id=remote_virtual_network)
+    )
+
+    if allow_virtual_network_access is None:
+        allow_virtual_network_access = True
```
Comment (fengzhou-msft) [inline src/databricks/azext_databricks/custom.py:119]: Set `default=True` in `_params.py`?

**Comment by fengzhou-msft** [comment]
History.rst also needs to be updated.

File: src/databricks/azext_databricks/commands.py (hunk 2 above)
Comment (bim-msft) [reply src/databricks/azext_databricks/commands.py:19]: @fengzhou-msft I saw the message from Veera "Yes after adding support for vNet Peering both PS and Az CLI can be released by removing experimental tags. ". I will double-check this with him.

File: src/databricks/azext_databricks/_params.py (hunk 1 above)
Comment (bim-msft) [reply src/databricks/azext_databricks/_params.py:60]: @fengzhou-msft This parameter only accepts ID now, I saw the parameter `custom_virtual_network_id` you added previously, does it support both name and ID? It seems this parameter will be passed into SDK function directly, no extra process found.

File: src/databricks/azext_databricks/custom.py (hunk 3 above)
Comment (bim-msft) [reply src/databricks/azext_databricks/custom.py:119]: @fengzhou-msft If I set `True` as default, then every time I execute `az databricks workspace vnet-peering update` will set `allow_virtual_network_access` to `True`, which is unexpected behavior. What I want is this default value only takes effect on `create`.

File: src/databricks/azext_databricks/_params.py (hunk 1 above)
Comment (fengzhou-msft) [reply src/databricks/azext_databricks/_params.py:60]: I added the support in a [validator](https://github.com/Azure/azure-cli-extensions/blob/7c7c086dfe6da58b0e87ce9eb0eb66deb056d8b4/src/databricks/azext_databricks/_validators.py#L33).

Comment (bim-msft) [reply src/databricks/azext_databricks/_params.py:60]: @fengzhou-msft Oh, I see. Good point.

**Comment by bim-msft** [comment]
azp run / rerun

File: src/databricks/azext_databricks/commands.py (hunk 2 above)
Comment (mvvsubbu) [reply src/databricks/azext_databricks/commands.py:19]: Yes you can remove Preview and release. However keep the Preview for CMK related options in documentation and help messages as discussed

**Pull request merged**